imports/plugins/included/jobcontrol/server/jobs/cart.js

```diff
@@ -0,0 +1,91 @@
+import later from "later";
+import moment from "moment";
+import { Accounts, Cart, Jobs } from "/lib/collections";
+import { Hooks, Logger, Reaction } from "/server/api";
+import { ServerSessions } from "/server/publications/collections/sessions";
+
+
+Hooks.Events.add("afterCoreInit", () => {
+  Logger.debug("Adding Job removeStaleCart and Accounts to jobControl");
+  const settings = Reaction.getShopSettings();
+  if (settings.cart) {
+    new Job(Jobs, "cart/removeFromCart", {})
+      .priority("normal")
+      .retry({
+        retries: 5,
+        wait: 60000,
+        backoff: "exponential" // delay by twice as long for each subsequent retry
+      })
+      .repeat({
+        schedule: later.parse.text("every day")
+      })
+      .save({
+        cancelRepeats: true
+      });
+  } else {
+    Logger.warn("No cart cleanup schedule");
+  }
+});
+
+/**
+ * {Function} that fetches stale carts
+ * @param {Object} olderThan older than date
+ * @return {Object} stale carts
+ */
+const staleCarts = (olderThan) => {
+  return Cart.find({ updatedAt: { $lte: olderThan } }).fetch();
+};
+
+export default () => {
+  const removeStaleCart = Jobs.processJobs("cart/removeFromCart", {
+    pollInterval: 60 * 60 * 1000, // backup polling, see observer below
+    workTimeout: 180 * 1000
+  }, (job, callback) => {
+    Logger.debug("Processing cart/removeFromCart");
+    const settings = Reaction.getShopSettings();
+    if (settings.cart) {
+      const schedule = (settings.cart.cleanupDurationDays).match(/\d/);// configurable in shop settings
+      const olderThan = moment().subtract(Number(schedule[0]), "days")._d;
+      const carts = staleCarts(olderThan);
+      carts.forEach(cart => {
+        if (cart.items) {
```

Comment:
So if I am reading this right, if the cart is empty we aren't deleting it? That definitely not what we want to do.